client_options.go

```diff
@@ -3,68 +3,105 @@ package walletclient
 import (
 	"net/http"
 
-	"github.com/bitcoin-sv/spv-wallet-go-client/transports"
+	"github.com/bitcoinschema/go-bitcoin/v2"
+	"github.com/libsv/go-bk/bec"
+	"github.com/libsv/go-bk/wif"
+	"github.com/pkg/errors"
 )
 
-// WithXPriv will set xPrivString on the client
-func WithXPriv(xPrivString string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.xPrivString = xPrivString
-		}
-	}
+// WalletClientConfigurator is the interface for configuring WalletClient
+type WalletClientConfigurator interface {
```

Comment:
Two thoughts:
- Since configurators are not used as inputs for public constructors - they perhaps  can be private
- Because the name of the package is `walletclient` I would skip the prefix and leave `Configurator` only.